elmPackages: add node packages

ursi opened the pull request:
Added:
- elm-install
- elm-git-install

Other node packages have also been updated via node2nix

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

ehmry reviewed (approved):
I don't know Elm but looks harmless to me.

ursi commented:
oof, the spacing is off

ursi commented:
Okay how can I fix this?

ehmry commented:
Rebase on the current master and force push.

ursi committed:
elmPackages: add node packages

Added:
- elm-install
- elm-git-install

Other node packages have also been updated via node2nix

ursi commented:
@ehmry I'd like to not have this sit here so long that I have to do this again. Is there anything I can do to up the chances of getting it merged?

turboMaCk commented:
I'm going to look into this PR

turboMaCk commented on pkgs/development/compilers/elm/packages/node-packages.json:
```diff
@@ -9,5 +9,7 @@
     "elm-xref",
     "create-elm-app",
-    "elm-optimize-level-2"
+    "elm-optimize-level-2",
```
this package was added in https://github.com/NixOS/nixpkgs/pull/96594

ursi replied:
What exactly are you getting at here?

turboMaCk replied:
just cleaning up old PRs. This one went under the carpet since no committer spotted it. Since then elm-optimize-level-2 was added. This adds elm-git-install https://github.com/NixOS/nixpkgs/pull/150458. That means `elm-install` is the only package from this PR which is not merged. I wonder - do you still want to include it in nixpkgs?

ursi replied:
> do you still want to include it in nixpkgs?

I don't see myself writing elm any time soon, so I'm kinda indifferent about it at this point. But if I don't have to do much to get it in, then yeah, why not.

turboMaCk commented:
I'm going to close this PR since it's outdated. Anyway @ursi feel free to ping me directly under your future PRs to nixpkgs. I might be able to get some of them in.

ursi commented:
@turboMaCk thanks!
